jmxtrans-output/jmxtrans-output-influxdb/src/main/java/com/googlecode/jmxtrans/model/output/InfluxDbWriter.java

```diff
@@ -0,0 +1,307 @@
+
+	/**
+	 * The names of the tag written to every {@link Point} that will contain the
+	 * value of {@link Server#getHost()}
+	 */
+	public static final String TAG_HOSTNAME = "hostname";
+
+	/**
+	 * Logger.
+	 */
+	private static final Logger LOG = LoggerFactory.getLogger(InfluxDbWriter.class);
+
+	/**
+	 * <p>
+	 * The name of the setting that can be used to control the write consistency
+	 * of each {@link BatchPoints} sent to InfluxDB. Allowed values are:
+	 * </p>
+	 * 
+	 * <ul>
+	 * <li>ALL = Write succeeds only if write reached all cluster members.</li>
+	 * <li>ANY = Write succeeds if write reached any cluster members.</li>
+	 * <li>ONE = Write succeeds if write reached at least one cluster members.
+	 * </li>
+	 * <li>QUORUM = Write succeeds only if write reached a quorum of cluster
+	 * members.</li>
+	 * </ul>
+	 *
+	 */
+	public static final String SETTING_WRITE_CONSISTENCY = "writeConsistency";
+
+	/**
+	 * <p>
+	 * The default value of write consistency for each measurement where no
+	 * writeConsistency setting is provided in the json config.
+	 * </p>
+	 * ALL = Write succeeds only if write reached all cluster members.
+	 */
+	private static final String DEFAULT_WRITE_CONSISTENCY = "ALL";
+
+	/**
+	 * 
+	 * The name of the setting that can be used to control the <a href=
+	 * "https://influxdb.com/docs/v0.9/concepts/key_concepts.html#retention-policy">
+	 * The retention policy</a> for the measurement
+	 */
+	public static final String SETTING_RETENTION_POLICY = "retentionPolicy";
+
+	/**
+	 * The deault <a href=
+	 * "https://influxdb.com/docs/v0.9/concepts/key_concepts.html#retention-policy">
+	 * The retention policy</a> for each measuremen where no retentionPolicy
+	 * setting is provided in the json config
+	 */
+	private static final String DEFAULT_RETENTION_POLICY = "default";
+
+	private String database;
+	private ConsistencyLevel writeConsistency = ConsistencyLevel.ALL;
+	private String retentionPolicy;
+
+	/** Thread safe **/
+	InfluxDB influxDB;
+
+	/**
+	 * @param typeNames
+	 * @param booleanAsNumber
+	 * @param debugEnabled
+	 * @param url
+	 *            - The url e.g http://localhost:8086 to InfluxDB
+	 * @param username
+	 *            - The username for InfluxDB
+	 * @param password
+	 *            - The password for InfluxDB
+	 * @param database
+	 *            - The name of the database (created if does not exist) on
+	 *            InfluxDB to write the measurements to
+	 * @param settings
+	 */
+	@JsonCreator
+	public InfluxDbWriter(@JsonProperty("typeNames") ImmutableList<String> typeNames,
+			@JsonProperty("booleanAsNumber") boolean booleanAsNumber, @JsonProperty("debug") Boolean debugEnabled,
+			@JsonProperty("url") String url, @JsonProperty("username") String username,
+			@JsonProperty("password") String password, @JsonProperty("database") String database,
+			@JsonProperty("settings") Map<String, Object> settings) {
```

Comment:
The use of the `settings` parameter is actually deprecated. Please pass all parameters explicitly to the constructor. (no, this is not really documented as it should, sorry).